parsers/lib/config.py

```diff
@@ -1,15 +1,19 @@
 import functools
 import os
+from collections.abc import ItemsView, KeysView, ValuesView
 from copy import deepcopy
 from datetime import timedelta
 from enum import Enum
 from inspect import signature
 from logging import getLogger
+from typing import TypeVar
 
 import requests
 from requests import Session
 from requests.adapters import HTTPAdapter, Retry
 
+ModeEnumType = TypeVar("ModeEnumType", bound="BaseModeEnum")
```

Comment:
This combined with the type changes below ensure that the right type is returned even when we use derived classes/enum.

Reply:
awesome, practicing type driven development :D